src/page-analysis/background/make-screenshot.js

```diff
@@ -8,7 +8,7 @@ import { dataURLToBlob } from 'blob-util'
 async function snapNow({ tabId }) {
     const tab = await browser.tabs.get(tabId)
     let image = await browser.tabs.captureVisibleTab(tab.windowId, {
-        format: 'png',
+        format: 'jpeg',
```

Comment:
If switching to JPEG here, YSK you can also pass a quality param which I found to have pretty good results. Not sure if better or worse than the resizing-by-canvas method you're using now though. Main difference should be  smaller dimensions (current method) vs JPEG artefacts. 
Docs: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/tabs/captureVisibleTab